lib/rules/declaration-property-unit-allowed-list/index.js

```diff
@@ -18,12 +19,24 @@ const messages = ruleMessages(ruleName, {
 	rejected: (property, unit) => `Unexpected unit "${unit}" for property "${property}"`,
 });
 
-function rule(list) {
+function rule(list, options) {
 	return (root, result) => {
-		const validOptions = validateOptions(result, ruleName, {
-			actual: list,
-			possible: [_.isObject],
-		});
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{
+				actual: list,
+				possible: [_.isObject],
+			},
+			{
+				actual: options,
+				possible: {
+					ignoreProperties: [_.isString, _.isRegExp],
```

Comment:
[suggestion] `ignoreProperties` seems needless to me.
(I'm not sure why the `color-named` rule has the `ignoreProperties` property... 🤷 )

```suggestion
```